1-org/modules/centralized-logging/main.tf

```
@@ -0,0 +1,188 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  key_first_resource         = keys(var.resources)[0]
+  logbucket_sink_member      = { for k, v in var.resources : k => v if k != var.logging_project_key }
+  resource_name              = var.logging_target_type == "bigquery" ? module.destination_bigquery[0].resource_name : var.logging_target_type == "pubsub" ? module.destination_pubsub[0].resource_name : var.logging_target_type == "storage" ? module.destination_storage[0].resource_name : module.destination_logbucket[0].resource_name
```

```suggestion
resource_map = {
  "bigquery" = module.destination_bigquery[0].resource_name ,
  "pubsub" = module.destination_pubsub[0].resource_name,
  "storage" = module.destination_storage[0].resource_name,
}
  resource_name = lookup(resource_map, var.logging_target_type, module.destination_logbucket[0].resource_name ) 
```

does this one works?

---

Done